### PR TITLE
Add RelEng packaging containers

### PR DESCRIPTION
**Pull request by coderobe**
resolves https://github.com/paritytech/ci_cd/issues/414
resolves https://github.com/paritytech/ci_cd/issues/415

see also https://github.com/paritytech/cleanroom/pull/26

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -190,6 +190,21 @@ node-wrk:
...
+rpm:
+  <<:                              *docker_build
+  script:
+    - *push_to_docker_hub
+
+deb:
+  <<:                              *docker_build
+  script:
+    - *push_to_docker_hub
+
```
Comment (chevdor) [inline .gitlab-ci.yml:207]: Should we prefix those with `cleanroom` or `rel` for instance @TriplEight ?

File: dockerfiles/gnupg/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+RUN apt-get update && apt-get install -yq --no-install-recommends bash ca-certificates curl gnupg
+
+RUN set -x \
+    && groupadd -g 1000 nonroot \
+    && useradd -u 1000 -g 1000 -s /bin/bash -m nonroot
+
+USER nonroot:nonroot
+
+# add gpg pubkey 'Parity Security Team <security@parity.io>'
+RUN curl -LfSs "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x9D4B2B6EB8F97156D19669A9FF0812D491B96798&options=mr&exact=on" | gpg --import - \
```
Comment (chevdor) [inline dockerfiles/gnupg/Dockerfile:28]: Could you bring the key to the top as ARG or ENV at least ?

**Review by chevdor — APPROVED** [review]
LGTM besides minor nits and the question wheter it works "as is" in R/O

**Review by chevdor — APPROVED** [review]
OK on my end

**Review by EgorPopelyaev — APPROVED** [review]

File: dockerfiles/gnupg/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# 'Parity Security Team <security@parity.io>'
+ARG GPG_KEYID=9D4B2B6EB8F97156D19669A9FF0812D491B96798
+
+FROM docker.io/library/ubuntu:latest
+
+# metadata
+LABEL summary="Base image for GnuPG operations" \
+	name="${REGISTRY_PATH}/gnupg" \
+	maintainer="devops-team@parity.io" \
```
Comment (TriplEight) [inline dockerfiles/gnupg/Dockerfile:13]: you could put L5 here

File: dockerfiles/gnupg/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	version="1.0" \
+	description="GnuPG base container" \
+	io.parity.image.vendor="Parity Technologies" \
+	io.parity.image.source="https://github.com/paritytech/scripts/blob/${VCS_REF}/dockerfiles/gnupg/Dockerfile" \
+	io.parity.image.documentation="https://github.com/paritytech/scripts/blob/${VCS_REF}/dockerfiles/gnupg/README.md" \
```
Comment (TriplEight) [inline dockerfiles/gnupg/Dockerfile:18]: please consider adding small READMEs, with general info what's there and why it's needed. We will be upstreaming them to dockerhub image descriptions.

File: .gitlab-ci.yml (hunk 1 above)
Comment (TriplEight) [reply .gitlab-ci.yml:207]: nah it looks fine, these images can be used out there, so generic names are better. I'd suggest prefixing them with the distro name/codename though.

**Review by TriplEight — CHANGES_REQUESTED** [review]
Also please mention
- do you want them rebuilt on schedule or manually?
- how often do you want to rebuild them?

File: dockerfiles/gnupg/Dockerfile (hunk 3 above)
Comment (coderobe) [reply dockerfiles/gnupg/Dockerfile:13]: L5?

**Comment by coderobe** [comment]
> do you want them rebuilt on schedule or manually?

manually works for the time being i think

File: dockerfiles/gnupg/Dockerfile (hunk 3 above)
Comment (chevdor) [reply dockerfiles/gnupg/Dockerfile:13]: Line 5 :)

**Comment by chevdor** [comment]
> Also please mention
> 
> * do you want them rebuilt on schedule or manually?
> * how often do you want to rebuild them?

Those should be rebuilt on schedule so we keep in sync with the GPG keys updates.
Ideally we should have them rebuilt on schedule with a low frequency such as one a week/month but also offer the option to rebuild them manually.

**Review by TriplEight — APPROVED** [review]

**Pull request merged**